src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeLift.java

```diff
@@ -124,6 +121,10 @@ private BytecodeLift(Block.Builder entryBlock, MethodModel methodModel, Value...
                         smfi -> entryBlock.block(smfi.stack().stream().map(BytecodeLift::toTypeElement).toList())))).orElse(Map.of());
 
         MethodTypeDesc mtd = methodModel.methodTypeSymbol();
+        ClassDesc thisClass = methodModel.parent().orElseThrow().thisClass().asSymbol();
+        if (!methodModel.flags().has(AccessFlag.STATIC)) {
+            mtd = mtd.insertParameterTypes(0, thisClass);
+        }
```

Comment:
There is duplication here and in the static method. It's minor but suggests the layering is awkward. I think it could be resolved if the private instance method `BytecodeLift::lift` returned the function operation, which is then further transformed for slots.

Reply:
I've reorganized and simplified `BytecodeLift` and `LocalsTypeMapper` parameters, deduplicated some calculations and separated the `liftToSlots` private method.
More changes might be required with extending the MH support in the next round.